File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        
        List<Integer> ans = new ArrayList<>(); 
        if(root==null) return ans;
        Queue<TreeNode> q = new ArrayDeque<>();
        
        q.offer(root);
        while(!q.isEmpty()){
            
            int n = q.size();
            for(int i=0; i<n; i++){
                TreeNode temp= q.poll();                
               
                if(temp.left!=null)
                    q.offer(temp.left);
                if(temp.right!=null)
                    q.offer(temp.right);
                if(i==n-1)
                    ans.add(temp.val);
            }
            
            
        }
        return ans;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self,root,level,list):
        if not root: return
        if level == len(list):
            list.append(root.val)
        self.dfs(root.right,level+1,list)
        self.dfs(root.left,level+1,list)
        
        
        
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        
        list = []
        
        self.dfs(root,0,list)
        
        return list
        
        
        